README.md

```diff
@@ -61,6 +61,7 @@ Returns a new focus trap on `element`.
 - **escapeDeactivates** {boolean}: Default: `true`. If `false`, the `Escape` key will not trigger deactivation of the focus trap. This can be useful if you want to force the user to make a decision instead of allowing an easy way out.
 - **clickOutsideDeactivates** {boolean}: Default: `false`. If `true`, a click outside the focus trap will deactivate the focus trap and allow the click event to do its thing.
```

Comment:
@gejgalis: Thanks for your patience. I think I understand your goals here and agree that it's useful. I have one source of hesitation: I worry that the relationship of `clickOutsideDeactivates` and `includeOutsideElement` is kind of confusing. What do you think about changing this up a little so we do not add an `includeOutsideElement` option but instead allow `clickOutsideDeactivates` to be a function? These would be the rules for `clickOutsideDeactivates`:

- If `true`, all clicks outside the focus trap deactivate it.
- If `false`, no clicks outside the focus trap deactivate it.
- If a function, when there's a click outside the focus trap the clicked element is passed to the function and a boolean is returned. If the function returns `true`, the focus trap is deactivated. If the function returns `false, the focus trap is not.

I'm thinking the advantage is that we no longer have a conflict between two options: `clickOutsideDeactivates` is the only one you set to control what outside clicks do.

How does that sound?

Reply:
I'm not sure if your proposal solves my problem. I don't want to activate/deactivate focus trap when clicked outside - I just want to skip preventing click events outside.

I think your previous proposal was better: istead of adding new option `includeOutsideElement` we could allow to setup multiple elements? What do you think @davidtheclark ?

Reply:
> I'm not sure if your proposal solves my problem.

@gejgalis I don't understand how it doesn't? — maybe the weirdness of the Kendo UI is confusing me. I thought the goal is to allow you to click on (but not focus) *some* external elements (the datepicker) without deactivating the focus trap? If so, I think it makes sense to say, essentially, "clickOutsideDeactivates when you click anywhere on the screen except within the datepicker" — and that's what a filter function could allow for.

Reply:
@davidtheclark I'm convinced now. It makes sense and it should work for me. You want me to prepare PR?

Reply:
Glad it made sense to you. Yeah, a PR would be great, @gejgalis!